### PR TITLE
MAINT: Enable vulture

### PR DESCRIPTION
**Pull request by larsoner**
Not 100% sure on adding this as a pre-commit hook since it takes 3-4s to run. `mypy` is similarly slow.

@drammock do you think it's okay to make `mypy` and `vulture` just run on CIs rather than as pre-commits? I dislike that commits already take so long due to `mypy` and with `vulture` the situation gets worse. These two will hopefully rarely trigger errors by changes anyway, and it's easy enough to run `vulture` or `mypy mne/evoked.py` directly if you're working on relevant code bits. And of course the CIs will complain if you don't run these and create problems. Draft mode for now until we decide.

This PR so far sets min confidence level 70% for ease of review. 60% confidence is much more useful because it catches some stuff we don't use (including the `mne/report` stuff we just removed in another PR!) but is also trickier because a lot of stuff needs to be added to the `allowlist` and the diff will get uglier. So better to sort out the initial integration details in this PR and then have a follow-up for 60% confidence level later.

### CODE REVIEW

File: mne/viz/_brain/_brain.py [hunk 1]
```diff
@@ -298,7 +298,6 @@ def __init__(
         views="auto",
         *,
         offset="auto",
-        offscreen=False,
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mne/viz/_brain/_brain.py:301]: I figured the risk was sufficiently low given that it didn't actually do anything. But I can be less lazy, sure :)

**Comment by hoechenberger** [comment]
@larsoner If speed is a concern, we could also wait for Ruff to implement this; but there's no ETA available today as far as I can tell

File: mne/time_frequency/tests/test_stft.py [hunk 2]
```diff
@@ -21,7 +23,7 @@
 def test_stft(T, wsize, tstep, f):
     """Test stft and istft tight frame property."""
     sfreq = 1000.0  # Hz
-    if True:  # just to minimize diff
+    with nullcontext():  # just to minimize diff
```
[the remark this answers is not in the document]
Comment (hoechenberger) [reply mne/time_frequency/tests/test_stft.py:26]: Agreed!

**Comment by drammock** [comment]
> If speed is a concern, we could also wait for Ruff to implement this; but there's no ETA available today as far as I can tell

I'm +1 on going ahead with vulture now (on CIs only) and when Ruff does support this, switching to Ruff (and enabling the precommit hook at that time)

**Comment by hoechenberger** [comment]
ok with me!

File: pyproject.toml [hunk 3]
```diff
@@ -309,6 +310,10 @@ ignore_messages = "^.*(Unknown target name|Undefined substitution referenced)[^`
 ignore_errors = true
 scripts_are_modules = true
 strict = false
+modules = ["mne"]
+# Avoid the conflict between mne/__init__.py and mne/__init__.pyi by ignoring the former
+exclude = '^mne/(beamformer|channels|commands|datasets|decoding|export|forward|gui|html_templates|inverse_sparse|io|minimum_norm|preprocessing|report|simulation|source_space|stats|time_frequency|utils|viz)?/?__init__\.py$'
+additional_dependencies = ["numpy==1.26.2"]
```
Comment (hoechenberger) [inline pyproject.toml]: Why is this needed?

Comment (larsoner) [reply pyproject.toml]: That was cruft from the pre-commit hook and it's gone now

File: pyproject.toml [hunk 4]
```diff
@@ -309,6 +310,13 @@ ignore_messages = "^.*(Unknown target name|Undefined substitution referenced)[^`
 ignore_errors = true
 scripts_are_modules = true
 strict = false
+modules = ["mne"]
+# Avoid the conflict between mne/__init__.py and mne/__init__.pyi by ignoring the former
+exclude = '^mne/(beamformer|channels|commands|datasets|decoding|export|forward|gui|html_templates|inverse_sparse|io|minimum_norm|preprocessing|report|simulation|source_space|stats|time_frequency|utils|viz)?/?__init__\.py$'
+
+[[tool.mypy.overrides]]
+module = ['scipy.*']
+ignore_missing_imports = true
```
Comment (larsoner) [inline pyproject.toml:319]: @hoechenberger did have to add this because of this failure, though

https://github.com/mne-tools/mne-python/actions/runs/8805892959/job/24169465941?pr=12569#step:6:12

With this change things are okay:

https://github.com/mne-tools/mne-python/actions/runs/8805919692/job/24169549047?pr=12569#step:6:12

Not sure why it happens on GHA but not Azure. Azure installs a lot more stuff and runs 3.11 instead of 3.12, maybe there's something wrong with the type stubs in SciPy or something for 3.12 (?). In any case I think this should be safe enough?

Comment (hoechenberger) [reply pyproject.toml:319]: I don't have the capacity to look into this now, but if it works, then let's just keep it like this for now :)
